src/ring_theory/ideal/prod.lean

```diff
@@ -68,8 +68,8 @@ end
 @[simp] lemma map_prod_comm_prod :
   map ↑(ring_equiv.prod_comm : R × S ≃+* S × R) (prod I J) = prod J I :=
 begin
-  rw [ideal_prod_eq (map _ _)],
-  simp [map_map]
+  rw [@ideal_prod_eq S R _ _ (map _ _)],
+  simp [map_map],
```

Comment:
Here's another way to help the elaborator:
```suggestion
  refine trans (ideal_prod_eq _) _,
  simp [map_map]
```

Reply:
(If we change enough things to `comm_semiring`, I think we won't need the `@` anymore. So maybe these suggestions are relatively pointless.)